src/core/support/slice_buffer.c

```diff
@@ -162,14 +163,30 @@ void gpr_slice_buffer_reset_and_unref(gpr_slice_buffer *sb) {
 }
 
 void gpr_slice_buffer_swap(gpr_slice_buffer *a, gpr_slice_buffer *b) {
-  gpr_slice_buffer temp = *a;
-  *a = *b;
-  *b = temp;
-
-  if (a->slices == b->inlined) {
+  GPR_SWAP(size_t, a->count, b->count);
+  GPR_SWAP(size_t, a->capacity, b->capacity);
+  GPR_SWAP(size_t, a->length, b->length);
+
+  if (a->slices == a->inlined) {
```

Comment:
Would it though? It would only change the ordering of the trickiness, effectively swapping a and b everywhere on the memcpy size argument. As the code stands now the memcpy size argument should always be the size used in the destination buffer... with them flipped it would be the size of the source buffer.